### DISCUSSION TITLE
The reason why wdpost doesn't start

### DISCUSSION THREAD

**Discussion by de-authority** [filecoin-project/lotus#7619] [Storage Provider]
Many community members reported that wdpost did not start and did not have any error logs
see these thread in slack:
- https://filecoinproject.slack.com/archives/CR98WERRN/p1636453460305200
- https://filecoinproject.slack.com/archives/CPFTWMY7N/p1636518034142300

In fact, there are log reminders, and you will see logs like this
> 2021-10-19T22:27:14.695+0900 WARN storageminer storage/wdpost_sched.go:123 window post scheduler notifs channel closed

Most of the reasons for this are: when the lotus miner is running wdpost, the lotus miner node / Lotus daemon node is performing sealing tasks, resulting in system resource overload. The notifs channels of lotus miner and lotus daemon close and lose synchronization status, and lotus miner will not start wdpost

**Comment by potato1992** [comment]
I have this issue but my miner is not doing sealing work.
The software did not even report any details about it. I have to restart the miner to recover it. 
Garbage code.

**Reply by de-authority** [reply]
1: Do you monitor your system for resources utilization 
2: filter your log by keywords "notifs channel closed"

**Comment by potato1992** [comment]
> 1: Do you monitor your system for resources utilization 2: filter your log by keywords "notifs channel closed"

The CPU/disk/GPU usage is near zero, I am not doing sealing on this machine. It's a miner only machine. Yes I can see "notifs channel closed" in the log.

**Reply by de-authority** [reply]
So you may have a bad network connection between `lotus miner` and `lotus daemon` or your `lotus daemon` didn't run stably

**Comment by potato1992** [comment]
> So you may have a bad network connection between `lotus miner` and `lotus daemon` or your `lotus daemon` didn't run stably

`lotus daemon` and `lotus miner` were running on the same machine. They communicated locally. I do not know how to judge if I have a stable `lotus daemon`. But it seems that `lotus daemon` was behaving properly during that time, how to judge it?

I still believe that some bugs were introduced in version v1.13.0, previously I was using  v1.11.2 and ` v1.11.2` was doing everything very smooth for two months. I also have seen many caches for `lotus worker` (mainly PC1 and PC2 workers) in v1.13.0. Also, `lotus worker` in v1.13.0 seems to require more memory than that in `v1.11.2`. In `v1.11.2`, I can run three docker with 7 PC1 and 2 C2 with 1TB memory, but in `v1.13.0` I can only run 6 PC1 and 2 C2 on the same machine. (running 7 would cause the PC1 docker being killed by excessive memory used)

**Comment by potato1992** [comment]
It is happening again, log:
```
2021-12-07T05:22:08.624+0800	WARN	storageminer	storage/wdpost_sched.go:123	window post scheduler notifs channel closed
```
The CPU usage plot:

![Screenshot from 2021-12-08 21-03-56](https://user-images.githubusercontent.com/22472437/145332815-8d63c938-58c1-44cc-a768-d72d82e6c733.png)

Ram usage is also very low:

![Screenshot from 2021-12-08 21-10-09](https://user-images.githubusercontent.com/22472437/145333036-567ec0e8-518a-483c-a421-4d9bb95da9e7.png)

The CPU is epyc 7402, installed ram: 256GB.

@dayou5168
